composer/callbacks/early_stopper.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Early stopping callback."""
+
+from __future__ import annotations
+
+import logging
+from typing import Callable
+
+import numpy as np
+
+from composer.core import State
+from composer.core.callback import Callback
+from composer.loggers import Logger
+
+log = logging.getLogger(__name__)
+
+__all__ = ["EarlyStopper"]
+
+
+class EarlyStopper(Callback):
+    __doc__ = f"""Callback to halt training early.
+    """
+
+    def __init__(
+        self,
+        monitor: str,
+        eval_label: str,
+        comp: Callable = None,
```

Comment:
Need more clarification on this - since I don't know what type of metrics are tracked are, I'm not sure if I'd know what A and B are. Also not sure about combining them.

maybe new docs might help clarify?

Reply:
In that case, I think you can do `Callable[[Any, Any], bool]` -- it should take any two inputs and return a boolean.